packages/aws-cdk-lib/aws-s3-deployment/lib/source.ts

```diff
@@ -262,5 +258,33 @@ export class Source {
     };
   }
 
+  /**
+   * Process objects such that it escapes token output suitable for JSON output.
+   *
+   * @param scope Parent construct scope
+   * @returns Object with with tokens escaped for JSON output.
+   */
+  private static escapeTokens(scope: Construct, obj: any): any {
+    if (Token.isUnresolved(obj)) {
+      // Return tokens as numbers. This is a hack to prevent the JSON serializer to wrap this token as a string.
+      // Stack.toJsonString should take care of escaping the object output for JSON.
+      return Token.asNumber(Stack.of(scope).toJsonString(obj));
+    }
+
+    if (Array.isArray(obj)) {
+      return obj.map(v => Source.escapeTokens(scope, v));
+    }
```

Comment:
A better approach might be to have `escapeTokens` accept an array, and perform the operation for each. If there is a single object, then the array would contain a single object.

Reply:
There may be a misunderstanding, `escapeTokens` already accepts an array.